security/seccomp/seccomp.go

```
@@ -0,0 +1,141 @@
+// +build linux,cgo,seccomp
+
+package seccomp
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"syscall"
+
+	libseccomp "github.com/mheon/libseccomp/src/goseccomp"
+)
+
+var (
+	actAllow libseccomp.ScmpAction = libseccomp.ActAllow
+	actDeny  libseccomp.ScmpAction = libseccomp.ActErrno.SetReturnCode(int16(syscall.EPERM))
+)
+
+// Filters given syscalls in a container, preventing them from being used
+// Started in the container init process, and carried over to all child processes
+func InitSeccomp(config SeccompConfig) error {
+	if !config.Enable {
+		return nil
+	}
+
+	var defaultAction libseccomp.ScmpAction
+	if config.WhitelistToggle {
+		defaultAction = actDeny
+	} else {
+		defaultAction = actAllow
+	}
+
+	filter, err := libseccomp.NewFilter(defaultAction)
+	if err != nil {
+		return fmt.Errorf("Error creating filter: %s", err)
+	}
+
+	// Unset no new privs bit
+	if err = filter.SetNoNewPrivsBit(false); err != nil {
```

I prefer to use masked errors in such places, like:
```
if err :=
```